eng/scripts/splittestdependencies/generate-dependency-functions.ps1

```diff
@@ -0,0 +1,85 @@
+        $projGroup[$groupNum] = [System.Collections.ArrayList]::new()
+      }
+      $projGroup[$groupNum].Add($projects[$i]) > $null
+      $i += 1
+    }
+    return ,$projGroup
+}
+
+
+# Take the group contains the project list. Write each group into a project file which can be used by dotnet test command.
+# Return the array of project files
+function Write-Test-Dependency-Group-To-Files($ProjectGroups, $MatrixOutputFolder) {
+  if (!$ProjectGroups -or !$ProjectGroups[0]) {
+    Throw "There should be some project files in the group. Please check the given project list."
+  }
+  $null = New-Item -Path $MatrixOutputFolder -ItemType "directory" -Force 
+  $numOfGroups = $ProjectGroups.Count
+  $allOutputFiles = @()
+  for ($i = 0; $i -lt $numOfGroups; $i++) {
+    $projectFilePath = "Project_$i.props"
+    $null = New-Item -Path "$MatrixOutputFolder/$projectFilePath" -ItemType "file" -Force
+    # Retain the structure without ProjectReference.
+    $templateXml=[xml]'<Project></Project>'  
+    $projectNode = $templateXml.SelectNodes("/Project")
+    $itemGroupNode=$templateXml.CreateNode("element", "ItemGroup", "")
+    $projectNode.AppendChild($itemGroupNode) > $null
+    foreach($pkg in $ProjectGroups[$i]) {
+      $newElem = $templateXml.CreateNode("element", "ProjectReference", "")
+      $newElemAttr = $templateXml.CreateAttribute("Include")
+      $newElemAttr.InnerText = $pkg
+      $newElem.Attributes.Append($newElemAttr) > $null
+      $itemGroupNode.AppendChild($newElem) > $null
+    }
+    $null = $templateXml.Save("$MatrixOutputFolder/$projectFilePath")
+    $allOutputFiles += $projectFilePath
+  }
+  return ,$allOutputFiles
+}
+
+# Add new property in the platform-matrix json and assign the values of the project file paths.
+function Write-Project-Files-To-Matrix($ProjectFiles, $MatrixJsonPath, $MatrixOutputFolder, $ProjectFileConfigName) {
+  if (!(Test-Path $MatrixOutputFolder)) {
+    $null = New-Item -Path $MatrixOutputFolder -ItemType "directory" -Force
+  }
+  $platformJson = Get-Content $MatrixJsonPath | ConvertFrom-Json
+  $overrideFiles = New-Object PSObject
+  foreach ($projectFile in $ProjectFiles) {
+    $n = $projectFile -replace "Project_(\d+).props", '$1'
+    $PropertyOverride = New-Object -Typename PSCustomObject -Property @{
+      $($ProjectFileConfigName) = $ProjectFile
+    }
+    $overrideFiles | Add-Member -Name "DependencyGroup$n" -value $PropertyOverride -MemberType NoteProperty
+  }
+  # Add one project files into the "include" session for short-term solution. 
+  foreach ($include in $platformJson.include) {
+    if (!$overrideFiles) {
+      Throw "Need at least one project file for include property. Please adjust the NumberOfTestsPerJob for better split."
+    }
+    $firstElem = $overrideFiles.PSObject.Properties | Select-Object -First 1
+    $overrideFiles.PSObject.properties.remove($firstElem.Name)
+    $include | Add-Member -Name "OverrideFiles" -value $firstElem.Value -MemberType NoteProperty
```

Comment:
I would still like for the project file ordering to match up with the test file ordering, as discussed in my earlier comment.

Reply:
Done.